test/e2e/apps/statefulset.go

```diff
@@ -577,11 +578,42 @@ var _ = SIGDescribe("StatefulSet", func() {
 		framework.ConformanceIt("Scaling should happen in predictable order and halt if any stateful pod is unhealthy [Slow]", func() {
 			psLabels := klabels.Set(labels)
 			w := &cache.ListWatch{
+				ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+					options.LabelSelector = psLabels.AsSelector().String()
+					// Set ResourceVersion to "" to ensure we start at now and won't
+					// miss needed or unintetionally include some past events.
+					options.ResourceVersion = ""
+					return f.ClientSet.CoreV1().Pods(ns).List(context.TODO(), options)
+				},
```

Comment:
for this fix, prefer minimal change to the test (e.g. continue starting from pl.ResourceVersion instead of providing a lister)

Reply:
fair enough - done